src/pgwire/src/message.rs

```diff
@@ -440,6 +442,61 @@ impl Severity {
             Severity::Log => "LOG",
         }
     }
+
+    /// Checks if a message of a given severity level should be sent to a client.
+    ///
+    /// The ordering of severity levels used for client-level filtering differs from the
+    /// one used for server-side logging in two aspects: INFO messages are always sent,
+    /// and the LOG severity is considered as below NOTICE, while it is above ERROR for
+    /// server-side logs.
+    pub fn should_output_to_client(&self, minimum_client_severity: &CoordClientSeverity) -> bool {
+        match (minimum_client_severity, self) {
+            // INFO messages are always sent
+            (_, Severity::Info) => true,
+            (CoordClientSeverity::Error, Severity::Error | Severity::Fatal | Severity::Panic) => {
+                true
+            }
+            (
+                CoordClientSeverity::Warning,
+                Severity::Error | Severity::Fatal | Severity::Panic | Severity::Warning,
+            ) => true,
+            (
+                CoordClientSeverity::Notice,
+                Severity::Error
+                | Severity::Fatal
+                | Severity::Panic
+                | Severity::Warning
+                | Severity::Notice,
+            ) => true,
+            (
+                CoordClientSeverity::Log,
+                Severity::Error
+                | Severity::Fatal
+                | Severity::Panic
+                | Severity::Warning
+                | Severity::Notice
+                | Severity::Log,
+            ) => true,
+            (
+                CoordClientSeverity::Debug1
+                | CoordClientSeverity::Debug2
+                | CoordClientSeverity::Debug3
+                | CoordClientSeverity::Debug4
+                | CoordClientSeverity::Debug5,
+                _,
+            ) => true,
+
+            (
+                CoordClientSeverity::Error,
+                Severity::Warning | Severity::Notice | Severity::Log | Severity::Debug,
+            ) => false,
+            (CoordClientSeverity::Warning, Severity::Notice | Severity::Log | Severity::Debug) => {
+                false
+            }
+            (CoordClientSeverity::Notice, Severity::Log | Severity::Debug) => false,
+            (CoordClientSeverity::Log, Severity::Debug) => false,
```

Comment:
I'm somewhat unhappy with this pattern match, but it gets the job done and it is fully exhaustive. I had another version which used more wildcards, but it felt a bit too clever in a bad way. 

(I also had another version which actually implemented `PartialOrd` between the two severities, but that just felt like a huge overkill).

Reply:
I am in favor of implementing this ordering scheme, because this function is also quite hairy. Compare the postgres implementation: https://github.com/postgres/postgres/blob/01ad1c953032ea2b88893d6a2f50fb4a8fea2873/src/backend/utils/error/elog.c#L237 which uses a nice `>=` to decide.

The comments in that function also describe what to do during auth and lists some security motivation. We should match that, and think about what the security thing they mention is.

I'm ok keeping the current big match block for now and delaying or maybe never doing the ordering thing, but we should do the "ignore this setting during auth" thing.